main.go

```diff
@@ -84,19 +96,31 @@ func main() {
 
 func ProcessIncomingConnection(conn net.Conn) {
 	defer conn.Close()
-	buf := make([]byte, 512)
+
 	for {
 		log.Print("SERVER: Connection: Waiting for the client")
-		n, err := conn.Read(buf)
+
+		n, err := base.ReadNetStringFromStream(conn, 1024 * 1024)
+
 		if err != nil {
 			log.Printf("SERVER: Connection: Starting reading Error: %s", err)
 			break
 		}
 
-		log.Printf("SERVER: Connection: Received %q\n", string(buf[:n]))
+		var decoder Decoder
+		err = json.Unmarshal(n, &decoder)
+
+		if err != nil {
+			log.Fatalf("SERVER: Connection: Error while decoding JSRPC message. %s", err)
+		}
+
+		if decoder.Method == "icinga::Hello" {
+			log.Printf("SERVER: JsonRpcConnection: Received %q from the client\n", decoder.Method)
+		}
+
+		err = base.WriteNetStringToStream(conn, n)
 
-		n, err = conn.Write(buf[:n])
-		log.Printf("SERVER: Connection: Sending %d bytes", n)
+		log.Printf("SERVER: Connection: Sending %d bytes message back", unsafe.Sizeof(n))
```

Comment:
Is this how Icinga would handle that method?

Reply:
No, but now it's all about receiving.

Reply:
What's all about receiving? Icinga's _Hello_ handling? Yes. Yours? No.